frontend/src/views/components/Journey/Embed/Embed.component.jsx

```diff
@@ -36,12 +38,15 @@ const Embed = (props) => {
   } = props;
   const { locale } = useRouter();
   useEffect(() => {
-    if (!layersLoaded || layersLocaleLoaded !== locale) {
-      loadLayers(locale);
+    const siteScope = mapUrl.startsWith('http') && getSubdomainFromURL(mapUrl);
+    const subdomainIsDifferentThanLoaded =
+      layersLoadedSubdomain !== siteScope && (layersLoadedSubdomain || siteScope);
```

Comment:
That line is very confusing to me. I'm not sure to understand under which circumstances `subdomainIsDifferentThanLoaded` is `true`.

If I create a truth table, I get this:
<table>
<tr>
 <th><code>layersLoadedSubdomain</code>
 <th><code>siteScope</code>
 <th><code>subdomainIsDifferentThanLoaded</code>
<tr>
 <td><em>falsy</em>
 <td><em>falsy</em>
 <td><code>false</code>
<tr>
 <td><em>falsy</em>
 <td><em>string</em>
 <td><code>true</code>
<tr>
 <td><em>string</em>
 <td><em>falsy</em>
 <td><code>true</code>
<tr>
 <td><em>string</em>
 <td><em>string</em>
 <td><code>true</code> if the strings are different
</table>

… which seems to indicate that it could be simply written as `layersLoadedSubdomain !== siteScope`. What do you think? Am I missing something?

Reply:
Is just that siteScope could be undefined and layersLoadedSubdomain could be null so if we compare them they will still be different